lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php

```diff
@@ -973,4 +978,13 @@ public function getSchemaSearchPaths() : array
 
         return $match[2];
     }
+
+    private function ensureDatabase(?string $database) : string
+    {
+        if ($database === null) {
+            throw new InvalidArgumentException('Needs a database.');
```

Comment:
Later, it will need a special exception.

Reply:
I will go ahead and add this now.